x-pack/plugins/security_solution/public/flyout/left/components/user_details.tsx

```diff
@@ -0,0 +1,283 @@
+import { manageQuery } from '../../../common/components/page/manage_query';
+import { useObservedUserDetails } from '../../../explore/users/containers/users/observed_details';
+import { useUserRelatedHosts } from '../../../common/containers/related_entities/related_hosts';
+import { useMlCapabilities } from '../../../common/components/ml/hooks/use_ml_capabilities';
+import { getEmptyTagValue } from '../../../common/components/empty_value';
+import { USER_DETAILS_RELATED_HOSTS_TABLE_TEST_ID, USER_DETAILS_TEST_ID } from './test_ids';
+import { ENTITY_RISK_CLASSIFICATION } from '../../../explore/components/risk_score/translations';
+import { HOST_RISK_TOOLTIP } from '../../../explore/hosts/components/hosts_table/translations';
+import * as i18n from './translations';
+
+const USER_DETAILS_ID = 'entities-users-details';
+const RELATED_HOSTS_ID = 'entities-users-related-hosts';
+
+const UserOverviewManage = manageQuery(UserOverview);
+const RelatedHostsManage = manageQuery(InspectButtonContainer);
+
+export interface UserDetailsProps {
+  /**
+   * User name for the entities details
+   */
+  userName: string;
+  /**
+   * timestamp of alert or event
+   */
+  timestamp: string;
+}
+/**
+ * User details and related users, displayed in the document details expandable flyout left section under the Insights tab, Entities tab
+ */
+export const UserDetails: React.FC<UserDetailsProps> = ({ userName, timestamp }) => {
+  const { to, from, deleteQuery, setQuery, isInitializing } = useGlobalTime();
+  const { selectedPatterns } = useSourcererDataView();
+  const dispatch = useDispatch();
+  // create a unique, but stable (across re-renders) query id
+  const userDetailsQueryId = useMemo(() => `${USER_DETAILS_ID}-${uuid()}`, []);
+  const relatedHostsQueryId = useMemo(() => `${RELATED_HOSTS_ID}-${uuid()}`, []);
+  const isPlatinumOrTrialLicense = useMlCapabilities().isPlatinumOrTrialLicense;
+  const narrowDateRange = useCallback(
+    (score, interval) => {
+      const fromTo = scoreIntervalToDateTime(score, interval);
+      dispatch(
+        setAbsoluteRangeDatePicker({
+          id: InputsModelId.global,
+          from: fromTo.from,
+          to: fromTo.to,
+        })
+      );
+    },
+    [dispatch]
+  );
+
+  const [isUserLoading, { inspect, userDetails, refetch }] = useObservedUserDetails({
+    id: userDetailsQueryId,
+    startDate: from,
+    endDate: to,
+    userName,
+    indexNames: selectedPatterns,
+    skip: selectedPatterns.length === 0,
+  });
+
+  const {
+    loading: isRelatedHostLoading,
+    inspect: inspectRelatedHosts,
+    relatedHosts,
+    totalCount,
+    refetch: refetchRelatedHosts,
+  } = useUserRelatedHosts({
+    userName,
+    indexNames: selectedPatterns,
+    from: timestamp, // related hosts are hosts this user has successfully authenticated onto AFTER alert time
+    skip: selectedPatterns.length === 0,
+  });
+
+  const relatedHostsColumns: Array<EuiBasicTableColumn<RelatedHost>> = [
```

Comment:
can memoize this with isPlatinumOrTrialLicense as a dependency, and that way you don't have to create a new array and push to it on every render

Reply:
thanks for catching this 👍